src/galileo/handlers/openai_agents.py

```diff
@@ -0,0 +1,333 @@
+        self._flush_on_chain_end: bool = flush_on_chain_end
+
+    def on_trace_start(self, trace: tracing.Trace) -> None:
+        """Called when a trace starts. Starts a Galileo trace."""
+        try:
+            # Use trace name as input, similar to Langchain example's root node
+            self._galileo_logger.start_trace(input=trace.name or "Unnamed Agent Trace")
+            self._trace_started.add(trace.trace_id)
+            _logger.debug(f"Started Galileo trace for agent trace_id: {trace.trace_id}")
+        except Exception as e:
+            _logger.error(f"Galileo: Error in on_trace_start for trace_id {trace.trace_id}: {e}", exc_info=True)
+
+    def on_trace_end(self, trace: tracing.Trace) -> None:
+        """Called when a trace ends. Concludes the Galileo trace."""
+        if trace.trace_id not in self._trace_started:
+            _logger.warning(f"Galileo: Received on_trace_end for unknown trace_id: {trace.trace_id}")
+            return
+        try:
+            # Conclude the overall trace. Output might be hard to determine here,
+            # maybe set a status? The final span's output might be more relevant.
+            self._galileo_logger.conclude(output=f"{trace.name}")
+            self._trace_started.remove(trace.trace_id)
+            _logger.debug(f"Ended Galileo trace for agent trace_id: {trace.trace_id}")
+            # Optionally flush here, or rely on shutdown/force_flush
+            if self._flush_on_chain_end:
+                self._galileo_logger.flush()
+        except Exception as e:
+            _logger.error(f"Galileo: Error in on_trace_end for trace_id {trace.trace_id}: {e}", exc_info=True)
+
+    def _log_span_start(self, span: tracing.Span[Any]) -> None:
+        """Handles the start of any span type by logging to Galileo."""
+        name = _get_span_name(span)
+        metadata = {"agent_span_id": span.span_id, "agent_trace_id": span.trace_id}
+        if parent_id := getattr(span, "parent_id", None):
+            metadata["parent_span_id"] = parent_id
+
+        # Default input/output - specific handlers will override
+        input_data = None
+        tags = None  # Add tags if agent spans support them
+        created_at = datetime.fromisoformat(span.started_at) if span.started_at else None
+
+        # --- Map Agent Span Types to Galileo Span Types ---
+
+        if isinstance(span.span_data, tracing.AgentSpanData):
+            # Treat Agent steps as workflow spans
+            metadata["tools"] = serialize_to_str(span.span_data.tools)
+            metadata["handoffs"] = serialize_to_str(span.span_data.handoffs)
+            metadata["output_type"] = serialize_to_str(span.span_data.output_type)
+            self._galileo_logger.add_workflow_span(
+                name=name,
+                input="Agent Step Start",  # Input often not known at start
+                metadata=convert_to_string_dict(metadata),
+                tags=tags,
+                created_at=created_at,
+            )
```

Comment:
IMO we need to make smaller methods and invoke each one here based on span_data type, it's easy to unit-test and maintain.